### PR TITLE
Change to modern layout in options dialog

### PR DESCRIPTION
**Pull request by CendioOssman**
Classical tabs are very dated. They are also a practical problem as you get very limited in the numbers of tabs we can have, and how long the text can be on them.

### CODE REVIEW

**Comment by CendioOssman** [comment]
Before:

Screenshot from 2022-09-07 14-45-30

After:

Screenshot from 2022-09-07 14-45-21

**Comment by CendioOssman** [comment]
This is broken out of #1410 so that it can be merged and used for #1517.

**Comment by dmcmahill** [comment]
Thanks for doing this.  The new dialog looks much nicer.

File: vncviewer/OptionsDialog.cxx [hunk 1]
```diff
@@ -62,19 +63,18 @@ std::map<OptionsCallback*, void*> OptionsDialog::callbacks;
...
   int x, y;
   Fl_Button *button;
 
-  Fl_Tabs *tabs = new Fl_Tabs(OUTER_MARGIN, OUTER_MARGIN,
-                             w() - OUTER_MARGIN*2,
-                             h() - OUTER_MARGIN*2 - INNER_MARGIN - BUTTON_HEIGHT);
-
+  // Odd dimensions to get rid of extra borders
+  Fl_Notebook *notebook = new Fl_Notebook(-1, -1, w()+2,
+                                          h()+1 - INNER_MARGIN - BUTTON_HEIGHT - OUTER_MARGIN);
```
[the remark this answers is not in the document]
Comment (CendioOssman) [reply vncviewer/OptionsDialog.cxx]: It was named after GtkNotebook, which was the Gtk widget you classically had to use to build something like this. It is equivalent to Fl_Tabs, though.

A GtkStackSidebar seems to just be the left list. And a GtkStack seems to widget handling the content. 

Qt seems to have a similar setup, where you create a QStackedWidget and pair it with QListWidget.

So no direct equivalent in the FOSS world.

WinUI has something very similar, though, which they call NavigationView.

I don't have any particular strong preferences for naming here.

**Comment by CendioOssman** [comment]
> See the below screenshot, do we want some sort of delimiter between the content and the window titlebar? To my eyes it looks a bit odd and "cut off" with Windows 11's white color.

Yeah, that doesn't look nice. I'll have a look.

**Comment by CendioOssman** [comment]
It would look like this instead:

Skärmbild 2022-12-21 140039

I only changed things on Windows, as it looked odd to increase the border on the other platforms.

File: vncviewer/OptionsDialog.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (samhed) [reply vncviewer/OptionsDialog.cxx]: I see. I like NavigationView, but Notebook is fine too.

**Comment by samhed** [comment]
Nice, that looks better!

**Pull request merged**